Add string ctor to MemberNotNull

stephentoub opened the pull request:
With only the params string[] ctor, every usage of MemberNotNull in a CLSCompliant(true) assembly needs to have its warning suppressed.

cc: @jcouv, @terrajobst 

Relates to https://github.com/dotnet/roslyn/issues/41438

jcouv commented:
Let’s vet with LDM. We’ll need a compiler change to recognize these new constructors.

jcouv commented:
Corresponding compiler change: https://github.com/dotnet/roslyn/pull/42712

jcouv commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs:
```diff
@@ -127,21 +127,25 @@ sealed class DoesNotReturnIfAttribute : Attribute
+        /// <param name="member">
+        /// The field and property member that is promised to be not-null.
+        /// </param>
+        public MemberNotNullAttribute(string member) => Members = new string[] { member };
```
`MemberNotNullWhenAttribute` also needs a similar change.

stephentoub replied:
Fixed

jcouv reviewed (commented):
Done with review pass (iteration 1)

stephentoub committed:
Add string ctor to MemberNotNull{When}

With only the params string[] ctor, every usage of MemberNotNull{When} in a CLSCompliant(true) assembly needs to have its warning suppressed.

jcouv reviewed (approved):
LGTM Thanks
